packages/waku/src/lib/renderers/html-renderer.ts

```diff
@@ -262,18 +262,28 @@ export const renderHtml = async (
             get(_target, name: string) {
               if (isDev) {
                 // TODO too long, we need to refactor this logic
-                let file = filePath.slice(config.basePath.length);
-                file = file.split('?')[0]!;
-                file = file.startsWith('@fs/')
-                  ? file.slice('@fs'.length)
-                  : encodeFilePathToAbsolute(joinPath(opts.rootDir, file));
+                const file = (
+                  filePath.slice(config.basePath.length).startsWith('@fs/')
+                    ? filePath.slice((config.basePath + '@fs').length)
+                    : filePath
+                ).split('?')[0]!;
                 const wakuDist = joinPath(
                   fileURLToFilePath(import.meta.url),
                   '../../..',
                 );
-                if (file.startsWith(wakuDist)) {
+                const fileWithAbsolutePath =
+                  !file.startsWith(opts.rootDir) && !file.startsWith(wakuDist)
+                    ? encodeFilePathToAbsolute(joinPath(opts.rootDir, file))
+                    : file;
+                if (
+                  file.startsWith(wakuDist) ||
+                  fileWithAbsolutePath.startsWith(wakuDist)
```

Comment:
This part looks a little bit complicated and has room for improvement.

Reply:
it makes /user/project/src/comp.tsx into /src/comp.tsx

Reply:
Yeah, but this entire block can be refactored for more readability, no?

Reply:
Hm, I think we can remove L279.

Reply:
> Hm, I think we can remove L279.


5e1ae0d